fix(upgrade): correctly handle removal of running program on windows

esatterwhite opened the pull request:
Unlike unix based systems, on windows based systems you cannot remove a file if it is still runining. This would cause the upgrade command to fail when it would try to replace the command in place. The recommendation on windows platforms is to rename the file, and use another process to remove that file at a later time.

This update ths upgrade command to rename the running binary and store the location of that file. The next time the program runs it will attempt to remove that file.

Fixes: #36

esatterwhite committed:
fix(upgrade): correctly handle removal of running program on windows

Unlike unix based systems, on windows based systems you cannot remove
a file if it is still runining. This would cause the upgrade command to
fail when it would try to replace the command in place. The
recommendation on windows platforms is to rename the file, and use
another process to remove that file at a later time.

This update ths upgrade command to rename the running binary and store
the location of that file. The next time the program runs it will
attempt to remove that file.

Fixes: #36

mdeltito reviewed (dismissed):
looks good, suggestion is related to debug output and unimportant

esatterwhite committed:
fix(upgrade): account for versions strings sans v

while the version list prints the versions with a v infront of the
smever string, it is rather common to input just the semver string sans
the v. This can cause a case where we match the version according the
semver, but the actual release tag may not be found which can skirt
around some validation in the parent upgrade machinary.

This accounts for the input of a raw semver string and appends the v if
not present to get the right tag name

esatterwhite commented on @mzm/core/update/providers/github.ts:
```diff
@@ -88,9 +88,44 @@ export class GHError extends Error {
+    await Deno.remove(old_file)
+  }
 
+  debug('windows executable swap complete')
```
yep. moved it
